File: week2_Section_2/Memory_management.cpp
#include <iostream>
#include <string>

class Car {
public:
    std::string brand;
    int year;

    Car(std::string b, int y) : brand(b), year(y) {}
    void display() { std::cout << "Car: " << brand << " - " << year << std::endl; }
};

int main() {
    Car* myCar = new Car("Ford", 2020); // Dynamic allocation
    myCar->display();

    delete myCar; // Manual memory deallocation to prevent leaks
    myCar = nullptr; // Avoid dangling pointer

    return 0;
}

File: week2_Section_2/Memory_management.rs
struct Car {
    brand: String,
    year: u32,
}

fn create_car(brand: String, year: u32) -> Car {
    Car { brand, year } // Ownership moves to caller
}

fn main() {
    let my_car = create_car(String::from("Tesla"), 2022);
    println!("Car: {} - {}", my_car.brand, my_car.year);
} // Memory is automatically deallocated